### PR TITLE
ECAL MultiFit: symmetric Pulse Shape Covariance Matrix DB object

### PR DESCRIPTION
**Pull request by emanueledimarco**
This PR introduces a new object, EcalPulseSymmCovariance, which reduces the memory allocation because this covariance matrix is a symmetric one.
To leave backward compatibility with the old object (EcalPulseShapeCovariance) the object is new in CondFormats/EcalObjects.

The tools that refer to it (CondCore/EcalPlugins and CondTools/Ecal) are not duplicated, just moved to use the new object. 

A PR with its usage in the ECAL local reconstruction will follow when the GT will be ready. 
Everything is validated with a local sqlite file. 

@bendavid @argiro @cerminar @diguida @ggovi please also follow this.

### CODE REVIEW

**Comment by emanueledimarco** [comment]
including also @ferriff following this.

**Comment by mmusich** [comment]
@cmsbuild  please test

**Comment by emanueledimarco** [comment]
Why the tests were not triggered? @davidlange6

**Comment by smuzaffar** [comment]
@mmusich , there was an extra space in b/w cmsbuild and please that is why cms-bot did not recognize it a valid request.
Anyway, I have update the regexp to allow extra spaces. tests are now running

**Comment by mmusich** [comment]
@smuzaffar thanks for the clarification!

**Comment by emanueledimarco** [comment]
Just to add one comment: no changes are expected from this PR.

**Comment by ggovi** [comment]
@emanueledimarco
I've just realised that you are missing the entries in CondCore/Utilities/src/CondFormats.h, CondCore/Utilities/src/CondDBImport.cc, CondCore/Utilities/src/CondDBFetch.cc  for your new payload. Can you please add them? See for reference the table under: "Guideline for creating new packages under the Cond system" in the page:

https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideCondSystemLayout#Guideline_for_creating_new_packa

Thanks

**Comment by emanueledimarco** [comment]
@ggovi 
Done (and thanks for the guidelines, I wasn't aware of such a twiki).

**Comment by ggovi** [comment]
+1

**Comment by emanueledimarco** [comment]
These cannot be related, since I just __added__ a new object in the DB, but it is not used anywhere in the reco (yet). When this is integrated, a new PR of the ECAL local reconstruction will follow, using this new object.

**Comment by emanueledimarco** [comment]
@diguida @mmusich if everything is fine, can we sign this, so I can proceed with the upload of the new condition and the changes in the ECAL reconstruction to use it? Thanks.

**Comment by davidlange6** [comment]
@diguida @mmusich - any issues with this PR?

**Comment by mmusich** [comment]
@davidlange6 to me is fine, but @diguida would like to take a chance to review it

File: CondCore/EcalPlugins/src/plugin.cc [hunk 1]
```diff
@@ -210,3 +213,4 @@ REGISTER_PLUGIN(EcalTimeBiasCorrectionsRcd, EcalTimeBiasCorrections);
 REGISTER_PLUGIN(EcalSamplesCorrelationRcd, EcalSamplesCorrelation);
 REGISTER_PLUGIN(EcalPulseShapesRcd,EcalPulseShapes);
 REGISTER_PLUGIN(EcalPulseCovariancesRcd,EcalPulseCovariances);
+REGISTER_PLUGIN(EcalPulseSymmCovariancesRcd,EcalPulseSymmCovariances);
```
Comment (diguida) [inline CondCore/EcalPlugins/src/plugin.cc]: This should be changed to use the real type, not a `typedef`:

````
REGISTER_PLUGIN(EcalPulseSymmCovariancesRcd,EcalCondObjectContainer<EcalPulseSymmCovariance>);
````

**Comment by diguida** [comment]
@emanueledimarco everything is ok from the point of view of the code. There is only a technical change to be done, see https://github.com/cms-sw/cmssw/pull/10105#discussion_r35194066 for record management (BTW, this should be propagated to all other containers).
We can discuss on how to integrate this in the GT: I was abroad when you contacted me on Skype...

**Comment by argiro** [comment]
> On 22 Jul 2015, at 11:13, Salvatore Di Guida <notifications@github.com> wrote:
> 
> @emanueledimarco everything is ok from the point of view of the code. There is only a technical change to be done, see #10105 (comment) for record management (BTW, this should be propagated to all other containers).

Emanuele is on vacation this week, can you please merge this PR, and right after I will do the change for all records ? Or do we need to have a GT before we can merge ?

**Comment by diguida** [comment]
@argiro thanks for you prompt reply, and for volunteering in this change. This was announced in the AlCa meeting during the Physics Week, but then remained in my todo list. I can now follow up on this also with other stakeholders.
This PR contains only the definition of the object, and the registration in the `EventSetup`. No client code is affected. At this point, I propose the following:
1. we merge this PR;
2. you make another PR with all the changes in the `DataProxy` registration;
3. everything is combined in the next pre-release: this way, when @emanueledimarco is back, he can send the files to the dropbox.

What do you think?

**Comment by diguida** [comment]
+1
conditional approval: see https://github.com/cms-sw/cmssw/pull/10105#issuecomment-123643474 for the milestones.

**Comment by argiro** [comment]
> On 22 Jul 2015, at 11:39, Salvatore Di Guida <notifications@github.com> wrote:
> 
> @argiro thanks for you prompt reply, and for volunteering in this change. This was announced in the AlCa meeting during the Physics Week, but then remained in my todo list. I can now follow up on this also with other stakeholders.
> This PR contains only the definition of the object, and the registration in the EventSetup. No client code is affected. At this point, I propose the following:
> 1. we merge this PR;
> 2. you make another PR with all the changes in the DataProxy registration;
> 3. everything is combined in the next pre-release: this way, when @emanueledimarco is back, he can send the files to the dropbox.
> 
> What do you think?
> 

sure, do you want to give me exact instructions on 2. privately ?

Stefano

**Comment by diguida** [comment]
@argiro thanks, I will send you instructions privately asap!

**Comment by davidlange6** [comment]
+1

**Pull request merged**